### PR TITLE
Add release instructions

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

**Comment by Carreau** [comment]
@minrk am I missing anything for a release ?

File: docs/source/development.rst [hunk 1]
```diff
@@ -30,3 +30,43 @@ the bundled and minified Javacript by using the option
...
+This would ask you for confirmation before removing all untracked file. Make
+sure the ``dist/`` folder in particular is clean and avoid stale build from
+previous attempts.
+
+Run the following two commands to generate the universal ``wheel`` and the ``sdist``:
+
+.. code::
+
+    $ python setup.py sdist
+    $ python setup.py bdist_wheel --universal
```
Comment (minrk) [inline docs/source/development.rst]: --universal is unnecessary

File: docs/source/development.rst [hunk 2]
```diff
@@ -30,3 +30,43 @@ the bundled and minified Javacript by using the option
...
+
+.. note::
+
+    The above command will modify (at least) the following file :
+    ``notebook/static/base/js/namespace.js`` to make the notebook version
+    availlable from javascript.
```
Comment (jdfreder) [inline docs/source/development.rst]: s/availlable/available

File: docs/source/development.rst (hunk 1 above)
Comment (Carreau) [inline docs/source/development.rst]: done.

File: docs/source/development.rst (hunk 2 above)
Comment (Carreau) [inline docs/source/development.rst]: done.

Comment (minrk) [inline docs/source/development.rst]: This note shouldn't be true, and doesn't appear to be. Where are you getting this?

**Comment by Carreau** [comment]
> This note shouldn't be true, and doesn't appear to be. Where are you getting this?

Hum, I build the bdist and the sdist and my git folder was unclean with this change, so ai assume it was normal.

**Comment by minrk** [comment]
This is missing the version bump/tag:

- update version in `notebook/_version.py`
- run `python setup.py jsversion`
- `git commit -am "release $VERSION"`
- `git tag -am "release $VERSION" $VERSION`

**Comment by minrk** [comment]
Ah, that might have been because you updated and committed the real version, but `jsversion` didn't run until it was fired as part of `sdist/bdist`. That should be done before the commit/tag.

**Comment by Carreau** [comment]
ok. I had'nt tagged. Thanks

**Comment by Carreau** [comment]
Git tag take a `-am` ?

**Comment by Carreau** [comment]
updated.

File: docs/source/development.rst [hunk 3]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
 number of requests that the browser make to the server, but can allow to test
 Javascript file modification without going through the compilation step that
 can take up to 30 sec.
+
+
+Making a notebook release
+-------------------------
+
+Make sure you have followed the step above and have all the tools to genereate
```
Comment (willingc) [inline docs/source/development.rst]: generate

File: docs/source/development.rst [hunk 4]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+the minified javascript and css files. 
+
+Make sure the repository is clean of any file that could be problematic. 
+You can remove all non-tracked files with:
+
+.. code::
+
+    $ git clean -xfdi
+
+This would ask you for confirmation before removing all untracked file. Make
```
Comment (willingc) [inline docs/source/development.rst]: files

File: docs/source/development.rst [hunk 5]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+sure the ``dist/`` folder in particular is clean and avoid stale build from
```
Comment (willingc) [inline docs/source/development.rst]: ... folder ~~in particular~~ is clean and remove any stale files from previous build attempts.

File: docs/source/development.rst [hunk 6]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+previous attempts.
+
+Run the following two commands to generate the universal ``wheel`` and the ``sdist``:
```
Comment (willingc) [inline docs/source/development.rst]: ~~two~~

File: docs/source/development.rst [hunk 7]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+
+Update version number in ``notebook/_version.py``.
```
Comment (willingc) [inline docs/source/development.rst]: Perhaps number the specific steps in the workflow.

File: docs/source/development.rst [hunk 8]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+
+Run ``$ python setup.py jsversion``. It will modify (at least)
+``notebook/static/base/js/namespace.js`` to make the notebook version available
+from within JavaScript.
+
+Commit and tag the with the current version number:
```
Comment (willingc) [inline docs/source/development.rst]: Missing "release" between the and with.

File: docs/source/development.rst [hunk 9]
```diff
@@ -30,3 +30,53 @@ the bundled and minified Javacript by using the option
...
+You are now ready to build the ``sdist`` and ``wheel``:
+
+.. code::
+
+    $ python setup.py sdist
+    $ python setup.py bdist_wheel
+
+
+You can now test the ``wheel`` and the ``sdist`` locally before uploading to PyPI.
+Make sure to use twine to upload the archives over SSL.
```
Comment (willingc) [inline docs/source/development.rst]: Add link to twine:  https://github.com/pypa/twine

**Comment by willingc** [comment]
@Carreau :smiley: I'm glad this is documented. A few minor suggested changes.

**Comment by willingc** [comment]
@carreau :smile_cat: Thank you!

**Pull request merged**